src/components/MultiSelect.js

```diff
@@ -0,0 +1,51 @@
+import React, { useCallback, useMemo } from 'react'
+import propTypes from '@dhis2/prop-types'
+import { MultiSelectField, MultiSelectOption } from '@dhis2/ui-core'
+
+import { FieldAdapter } from './FieldAdapter.js'
+
+const useChangeHandler = onChange =>
+    useCallback(
+        ({ selected }) => {
+            const value = selected.map(({ value }) => value)
+            onChange({ value }, null)
+        },
+        [onChange]
+    )
+
+const useSelectedOptions = (options, value) =>
+    useMemo(() => {
+        const selectedValues = new Set(value || [])
+        return options.filter(({ value }) => selectedValues.has(value))
+    }, [options, value])
+
+const MultiSelectComponent = ({ options, value, onChange, ...rest }) => {
+    const handleChange = useChangeHandler(onChange)
+    const selectedOption = useSelectedOptions(options, value)
+
+    return (
+        <MultiSelectField
+            {...rest}
+            selected={selectedOption}
+            onChange={handleChange}
+        >
+            {Array.isArray(options)
+                ? options.map(option => (
+                      <MultiSelectOption key={option.value} {...option} />
+                  ))
+                : undefined}
+        </MultiSelectField>
+    )
+}
+
+MultiSelectComponent.propTypes = {
+    options: propTypes.arrayOf(MultiSelectField.propTypes.selected),
```

Comment:
If I see correctly, the `MultiSelectField.propTypes.selected` is already an array, it resolved to [this prop type](https://github.com/dhis2/ui-core/blob/f470ad74fbd81f1037954d1716af36215d8ff646/src/common-prop-types.js#L23).

Reply:
Well spotted, this was a difference between multi and single that I overlooked